File: src/main/kotlin/com/smody/book/book/api/response/NaverBookApiResponse.java
package com.smody.book.book.api.response;

import lombok.Getter;

@Getter
public class NaverBookApiResponse implements BookApiResponse {

    private String title;
    private String link;
    private String image;
    private int discount;
    private String publisher;
    private String pubdate;
    private String author;
    private String isbn;
    private String description;

    @Override
    public String title() {
        return title;
    }

    @Override
    public String image() {
        return image;
    }

    @Override
    public String author() {
        return author;
    }

    @Override
    public String pubdate() {
        return pubdate;
    }

    @Override
    public String publisher() {
        return publisher;
    }

    @Override
    public String description() {
        return description;
    }
}

File: src/main/kotlin/com/smody/book/book/api/naver/NaverBookApiResponse.kt
package com.smody.book.book.api.naver

import com.smody.book.book.api.BookApiResponse

class NaverBookApiResponse(
    val title: String,
    val link: String,
    val image: String,
    val discount: Int,
    val publisher: String,
    val pubdate: String,
    val author: String,
    val isbn: String,
    val description: String,
) : BookApiResponse {

    override fun title(): String {
        return title
    }

    override fun image(): String {
        return image
    }

    override fun author(): String {
        return author
    }

    override fun pubdate(): String {
        return pubdate
    }

    override fun publisher(): String {
        return publisher
    }

    override fun description(): String {
        return description
    }
}
